app/scripts/views/complete_sign_up.js

```diff
@@ -60,6 +63,22 @@ function (Cocktail, FormView, BaseView, CompleteSignUpTemplate,
       return self.fxaClient.verifyCode(uid, code)
           .then(function () {
             self.logScreenEvent('verification.success');
+            var account = self.getAccount();
+
+            if (account.get('needsOptedInToMarketingEmail')) {
+              account.unset('needsOptedInToMarketingEmail');
+              // TODO is the account automatically persisted?
+
+              var emailPrefs = account.getMarketingEmailPrefs();
+              return emailPrefs.optIn(NEWSLETTER_ID)
```

Comment:
Email opt-in is *really* slow. Should we skip waiting for the result and just hope for the best? If we do skip waiting, we have to ensure the XHR request is not aborted in the OAuth redirect flow. mmm... maybe wait.